src/v0.1.0/properties/structures/_cheminfo_smiles.yaml

```diff
@@ -0,0 +1,77 @@
+$$schema: "https://schemas.optimade.org/meta/v1.2/optimade/property_definition"
+$id: "https://schemas.optimade.org/namespaces/cheminformatics/v0.1/properties/structures/_cheminfo_smiles"
+title: "SMILES (Simplified Molecular Input Line Entry System) representation of the structure"
+x-optimade-type: "_cheminfo_smiles"
+x-optimade-definition:
+  kind: "property"
+  version: "0.1.0"
+  format: "1.2"
+  name: "_cheminfo_smiles"
+  label: "_cheminfo_smiles_structures"
+type:
+  - "string"
+  - "null"
+description: |-
+  SMILES (Simplified Molecular Input Line Entry System) representation of the structure.
+  Providers MAY canonicalize (i.e., use rules to establish the stable order of atoms) the produced SMILES representations, but this is not mandatory and no particular set of rules is recommended.
+  Generally, providers SHOULD NOT change the representation more frequently than the structure itself is modified.
+
+  In the filtering context, this field behaves differently to a normal string.
+  Firstly, this property can be used with the `CONTAINS` operator to provide a substructure query using the SMILES Arbitrary Target Specification (SMARTS) language defined in the Daylight SMARTS documentation (https://www.daylight.com/dayhtml/doc/theory/theory.smarts.html).
+  Secondly, exact matches using the `=` operator have additional semantics that allow a server to canonicalize the SMILES value before performing the exact match comparison, i.e., a server implementation SHOULD attempt to convert a user query for an exact SMILES match into the particular SMILES format used by the database, whereupon it can return entries that have SMILES data that matches at the semantic rather than lexical level.
+  Finally, `IS KNOWN` and `IS UNKNOWN` operators MAY be used to filter entries having non-`null` and `null` values of this property, accordingly.
+examples:
+  - "c1ccccc1"
+  - "CN1C=NC2=C1C(=O)N(C(=O)N2C)C"
+  - "[Na+].[Cl-]"
+x-optimade-metadata-definition:
+  title: "Metadata for the _cheminfo_smiles field"
+  description: "This dictionary contains the per-entry metadata for the _cheminfo_smiles field."
+  x-optimade-type: "dictionary"
+  type:
+    - "object"
+    - "null"
+  properties:
+    _cheminfo_specification_name:
+      description: |-
+        A name of the specification the given SMILES string conforms to.
+        Takes value from an enumeration set containing 'IUPAC SMILES+' (https://github.com/IUPAC/IUPAC_SMILES_plus), 'OpenSMILES' (http://opensmiles.org/opensmiles.html) and 'other' (none of the above).
+        If 'other' is chosen, the specification MUST be specified in metadata property '_cheminfo_specification_name_other'.
+      x-optimade-type: "string"
+      type:
+        - "string"
+        - "null"
+      enum:
+        - "IUPAC SMILES+"
+        - "OpenSMILES"
+        - "other"
+      examples:
+        - "IUPAC SMILES+"
+        - "OpenSMILES"
+      x-optimade-unit: "inapplicable"
+    _cheminfo_specification_name_other:
```

Comment:
I am happy to go with the "name_other" approach, however, it might be better to take the CIF approach and name the field "_cheminfo_specification_special_details". We could still encourage for this field to be filled only if the specification name is set to `other`, however, databases could also convey some additional details about the SMILES (e.g. "OpenSMILES specification is followed, but stereochemistry is not provided)".

Reply:
Personally I am not a big fan of `special_details`. I know that for people coming from CIF world this is intuitive, but for others (and for me sometimes) this looks as if free-form catch-all text field to fill with any details which do not fit anywhere else.

Reply:
Could we then introduce a separate catch-all field for the SMILES specification details? A name may be useful (e.g. referring to "Daylight SMILES" or "Chemaxon SMILES"), however, there may be more details regarding the SMILES treatment that a database may want to convey. For example, a database may use OpenSMILES, but completely avoid stereochemistry descriptions. In the case of the COD, we mainly follow the OpenSMILES standard, however, there is also a paper we could refer to that describes the depiction of certain compounds (organometallics, etc.).